### ISSUE TITLE
Images are not generated by custom size

### ISSUE THREAD

**Issue by dandyweng** [Bonobolabs/gatsby-remark-image-custom-widths#1]
Hello,

I really like the idea of how you made it possible to specify a size for individual images in Markdown, and as far as I know, this plugin seems to be the only viable way of achieving this. However, there appears to be an issue where the custom width does not affect the sizes of generated responsive images.

Here's my `gatsby-config`
```js
plugins: [
        `@bonobolabs/gatsby-remark-images-custom-widths`,
       {
            resolve: `gatsby-plugin-mdx`,
            options: {
                extensions: [`.mdx`, `.md`],
                remarkPlugins: [require(`remark-unwrap-images`)],
                gatsbyRemarkPlugins: [
                    {
                        resolve: `@bonobolabs/gatsby-remark-images-custom-widths`,
                        options: {
                            maxWidth: 1200,
                            withWebp: true,
                            showCaptions: [`title`, `alt`],
                            linkImagesToOriginal: false,
                        },
                    },
                ],
            },
        },
        // ...
]
```

And here in my MDX file I specified an image with a `380px` width
```html
<img src="./image.png" alt="Test image" width="380px" />
```

Generated HTML
```html
<figure class="gatsby-resp-image-figure">
  <span class="gatsby-resp-image-wrapper" style="/* ... */ max-width: 380px;"></span>
    <picture>
        <source srcset="/static/.../image.webp 300w,/static/.../image.webp 600w,/static/.../image.webp 746w" sizes="(max-width: 746px) 100vw, 746px" type="image/webp">
        <source srcset="/static/.../image.png 300w,/static/.../image.png 600w,/static/.../image.png 746w" sizes="(max-width: 746px) 100vw, 746px" type="image/png">
        <img class="gatsby-resp-image-image" src="/static/.../image.png" alt="Test image" title="Test image" loading="lazy" style="...">
      </picture>
  </span>
  <figcaption class="gatsby-resp-image-figcaption">Test image</figcaption>
</figure>
```

The `max-width: 380px;` CSS has been added to the wrapper's style so the plugin is running, but responsive sizes in the `<source>` tags are wrong. Here's what it looks like after I changed the `maxWidth` option to `380` in plugin's config file:

```html
<figure class="gatsby-resp-image-figure">
  <span class="gatsby-resp-image-wrapper" style="/* ... */ max-width: 380px;"></span>
    <picture>
      <source srcset="/static/.../image.webp 95w,/static/.../image.webp 190w,/static/.../image.webp 380w,/static/.../image.webp 570w,/static/.../image.webp 746w" sizes="(max-width: 380px) 100vw, 380px" type="image/webp">
      <source srcset="/static/.../image.png 95w,/static/.../image.png 190w,/static/.../image.png 380w,/static/.../image.png 570w,/static/.../image.png 746w" sizes="(max-width: 380px) 100vw, 380px" type="image/png">
      <img class="gatsby-resp-image-image" src="/static/.../image.png" alt="Test image" title="Test image" loading="lazy" style="...">
    </picture>
  </span>
  <figcaption class="gatsby-resp-image-figcaption">Test image</figcaption>
</figure>
```

I'm fairly new to Gatsby and MDX, so I'm not sure if I did something wrong, but it seems the plugin is not passing the custom width to the plugin's `maxWidth` option. If that's the case, I'd be thrilled if you can fix it, and I'm looking forward to more improvements to this plugin, e.g., the ability to add class names and custom styles to individual images.

Thanks!

**Comment by alexlouden** [comment]
Hey, thanks for the comment - glad you're finding this useful!

We're declaring it a bit differently for https://moleskinestudio.com/:

```js
  plugins: [
    {
      resolve: 'gatsby-plugin-mdx',
      options: {
        extensions: ['.mdx', '.md'],
        plugins: [
          '@bonobolabs/gatsby-remark-images-custom-widths',
          'gatsby-remark-autolink-headers'
        ],
        gatsbyRemarkPlugins: [
          {
            resolve: '@bonobolabs/gatsby-remark-images-custom-widths',
            options: {
              maxWidth: 740,
              linkImagesToOriginal: false,
              quality: 80,
              wrapperStyle: 'border-radius: 8px; overflow: hidden;'
            }
          },

```

Then for images in the MDX file (e.g. on https://moleskinestudio.com/support/timepage/widget/countdown/):
```js
<div class="side-by-side">
<div class="iphone portrait">
<img src="../assets/countdown_event.png" alt="Timepage event screen showing the countdown in 5 days for a flight to Melbourne." width="300px"/>
</div>
<div class="iphone portrait">
<img src="../assets/countdown_earnest.png" alt="Timepage Countdown screen showing a countdown of two days, 17 hours and 32 minutes until it’s time to call Ernest." width="300px"/>
</div>
</div>
```

This is with gatsby 2.17.11 and gatsby-plugin-mdx 1.0.55, not sure if it's different in more recent versions sorry!

**Comment by dandyweng** [comment]
Hi, thanks for your reply!

I think the issue is caused by [this commit](https://github.com/Bonobolabs/gatsby-remark-image-custom-widths/commit/929fd3d60d05022f669f0452d238bd08e9cb7c74). I downgraded your plugin to v0.0.3 and it's working as expected now. Can you explain why you decided not to mutate those options?

Thanks again.

**Comment by alexlouden** [comment]
Ah good find - yes, that's a bug I introduced, whoops!

I made that change to fix a bug where the size of one images would apply to the next images that were processed, e.g.:
```
<img src="image1.png" width="300px"/>
<img src="image2.png" /> // also 300px wide
```

I've just made a fix, and pushed 0.0.5 - would you mind trying that?
https://github.com/Bonobolabs/gatsby-remark-image-custom-widths/commit/daefe7b3262cc9ecdb5edf3640bdfa52bd68fe9f

**Comment by dandyweng** [comment]
Thanks for the fix! But it seems it's not working for webp images.

Generated HTML:
```html
<picture>
  <source srcset="/static/.../image.webp 300w,/static/.../image.webp 600w,/static/.../image.webp 746w" sizes="(max-width: 746px) 100vw, 746px" type="image/webp">
  <source srcset="/static/.../image.png 95w,/static/.../image.png 190w,/static/.../image.png 380w,/static/.../image.png 570w,/static/.../image.png 746w" sizes="(max-width: 380px) 100vw, 380px" type="image/png">
  <img class="gatsby-resp-image-image" src="/static/.../image.png" alt="Test Image" title="Test Image" loading="lazy" style="...">
</picture>
```

**Comment by alexlouden** [comment]
Ah, thanks for letting me know! I've fixed that in 0.0.6 :)

**Comment by dandyweng** [comment]
Great! I'm closing this issue now. I'd love to see more improvements in the future. It would be cool if we can specify classes like this:

```html
<img src="./image.jpg" alt="My image" width="800px" class="foo bar" />
```

and have them appended to `gatsby-resp-image-wrapper`. For now, I have to wrap a div around the image to apply different CSS styles:

```html
<div class="foo bar">
    <img src="./image.jpg" alt="My image" width="800px" />
</div>
```

It works, but I'm not too fond of having unnecessary divs in my markdown. I'm looking forward to the leaner way of achieving this.

Thanks again.
